### ISSUE TITLE
Audio/Video ingest: Create Condition to prevent generating Service Files, if they exist already

### ISSUE THREAD

**Issue by bseeger** [jhu-idc/iDC-general#427]
A Condition should be created that will check to see if an Node has a Service File associated with it already before it triggers the services to create one. 

This condition should be applied to the Audio and Video Derivative generation Contexts. This will help the deriv generating services (maybe just Homarus) by not asking it to create a derivative each time a file is ingested and only when we really need them to. 

It is unclear what, if anything, the derivatives were doing anyways to these files and they can take a very long time to create.  It's possible that creating a derivative does nothing for the user's experience (filesize-wise) anyways.

**Comment by mjanowiecki** [comment]
Does this mean in the migration spreadsheets that the service file needs to be listed before the original files?

**Comment by bseeger** [comment]
Hmmm, mostly likely "no", because hopefully it only saves the whole Media object when all the fields are ingested.  So Original File and Service File would be set at the same time.  It will be something to test out thought to ensure that's the case.

**Comment by birkland** [comment]
It is  currently re-encoding videos to a known format, mp4, with no other changes.  For original files that are already mp4s, it spends a lot of heat and time generating a derivative that is functionally identical.  For, say, an mpeg2 Original File, there is some value in converting it to an mp4 for a service file.  

More generally, it allows the system to normalize the characteristics of the videos it serves out.    See also the config.  We're not leveraging it, but theoretically we could specify a particular codec (a well-supported one), bounds on resolution or frame rate, use higher compression, etc.  

![image](https://user-images.githubusercontent.com/517881/135503944-b75b9b51-0106-4fcd-a92c-85d079e869ac.png)

**Comment by bseeger** [comment]
I've created a condition and see that it's working well, from what I can tell.  

@birkland, In testing what I'm noticing is that when a service file is created, it is smaller than the original. Not sure how much it's changed in quality though.   The file length, in seconds, is the same. 

For example one file I uploaded has a filesize of 9840497.  I had it create a service file and that file has a size of: 1599816. So there is some sort of optimization happening there.  

For reference, I used the 3MB and 10MB files from this page for testing: https://file-examples.com/index.php/sample-video-files/sample-mp4-files/

The command homarus is running is this: 

```
"ffmpeg  -i http://islandora-idc.traefik.me/system/files/2021-10/earth_mp4_1280_10mg.mp4  -ss 00:00:01.000 -frames 1 -vf scale=100:-2 -loglevel error  -f image2pipe -"
```
